src/main.rs

```diff
@@ -1,7 +1,8 @@
 use std::sync::Arc;
 
 use aws_config::meta::region::RegionProviderChain;
-use aws_sdk_s3::{ByteStream, Client, Region};
+use aws_sdk_s3::{ByteStream, Client, Region, Endpoint};
+use http::Uri;
```

Comment:
Please move this import after `futures` (we try to keep imports in alphabetical order)